### PR TITLE
!!! FEATURE: Adjust documentation and dependencies for new redirecthandler version

### PR DESCRIPTION
**Pull request by Sebobo**

### CODE REVIEW

File: Documentation/index.rst [hunk 1]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+
+
+It is also possible to add, change or remove redirects within the CLI.
+The available CLI commands for custom redirect management can be found in the `Neos Command Reference`.
+
+
+Restrict generation
+-------------------
+
+It is possible to restrict the generation of redirects to a certain node path or node type. For instance, you can user
```
Comment (jonnitto) [inline Documentation/index.rst]: The sentence `For instance, you can user…` isn't clear. I think there's a typo
```suggestion
It is possible to restrict the generation of redirects to a particular node path or node type. For instance, you can use it
```

File: Documentation/index.rst [hunk 2]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+
+**Note**: No redirect will be created if you move a node within the restricted path or if you move it away from the
+restricted path. But if you move a node into the restricted path the restriction rule will not apply, because the
+restriction is based on the source node path.
+
+```
+restrictByPathPrefix:
+  - '/sites/neosdemo': true
+```
+
```
Comment (jonnitto) [inline Documentation/index.rst]: This part below should be an own chapter as it doesn't not only to `Automatically generated redirects in Neos`

File: Documentation/index.rst [hunk 3]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+=========================================
+
+Whenever you change the `URL path segment` or move a document node, a redirect will automatically be generated as soon as it is published into the live workspace.
+
+.. note:: To get an overview over all currently active redirects you can always run ``./flow redirect:list``. For further details check the `Neos Command Reference`.
+
+Possible configuration for redirects
+------------------------------------
+
+You can configure the default behaviour for automatically generated redirects within ``Settings.yaml``.
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
You can configure the default behavior for automatically generated redirects within ``Settings.yaml``.
```

File: Documentation/index.rst [hunk 4]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
   restrictByOldUriPrefix:
     '/some/uri/path': true
+
+=========================================
+Automatically generated redirects in Neos
+=========================================
+
+Whenever you change the `URL path segment` or move a document node, a redirect will automatically be generated as soon as it is published into the live workspace.
+
+.. note:: To get an overview over all currently active redirects you can always run ``./flow redirect:list``. For further details check the `Neos Command Reference`.
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
.. note:: To get an overview over all currently active redirects you can always run ``./flow redirect:list``. For further details, check the `Neos Command Reference`.
```

File: Documentation/index.rst [hunk 5]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+Options
+^^^^^^^
+
+``hitCounter``
+  turn on/off the hit counter for redirects.
+``statusCode``
+  define the default status code for redirect or gone status (node deleted).
+
+
+It is also possible to add, change or remove redirects within the CLI.
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
It is also possible to add, change, or remove redirects within the CLI.
```

File: Documentation/index.rst [hunk 6]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+The available CLI commands for custom redirect management can be found in the `Neos Command Reference`.
+
+
+Restrict generation
+-------------------
+
+It is possible to restrict the generation of redirects to a certain node path or node type. For instance, you can user
+in an multi site environment or avoid massive redirect generation if you don't need it.
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
in a multi-site environment or avoid massive redirect generation if you don't need it.
```

File: Documentation/index.rst [hunk 7]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+
+The expected format is this:
+
+```
+"Source Uri","Target Uri","Status Code",Host,"Start DateTime","End DateTime",Comment,Creator,Type
+my-source-uri,my-target-uri,307,,2019-06-28-13-11-00,2019-06-30-13-11-00,"New product",myName,manual
+```
+
+It is recommended to first export your redirects and work on the resulting file, so you already have this structure.
+But you can also create a CSV file yourself. The first row with headings is optional as are all columns except the
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
However, you can also create a CSV file yourself. The first row with headings is optional as are all columns except the
```

File: Documentation/index.rst [hunk 8]
```diff
@@ -74,3 +74,135 @@ Restrict redirect generation by old URI prefix.
...
+first three `Source Uri`, `Target Uri` and `Status Code`.
+That means you can create a CSV file with 3 columns and without a header row and the import should work fine.
+If you use the other fields to, you must have them in the correct order or errors might occur.
+
+No deletions will be done, when the imported CSV file doesn't contain existing redirects.
+
+Deletions might occur when redirect chains are resolved. So a redirect from `A -> B` and a redirect from `B -> C` will be
+combined into one redirect `A -> C`.
+If you still need the redirect from `B -> C`, you might need to add that one afterwards again.
```
Comment (jonnitto) [inline Documentation/index.rst]:
```suggestion
If you still need the redirect from `B -> C`, you might need to add that one afterward again.
```

File: Documentation/index.rst (hunk 2 above)
Comment (Sebobo) [reply Documentation/index.rst]: Sorry don't fully understand what you mean?

File: Documentation/index.rst (hunk 5 above)
Comment (Sebobo) [reply Documentation/index.rst]: Here I disagree ;)

File: Documentation/index.rst (hunk 8 above)
Comment (Sebobo) [reply Documentation/index.rst]: `afterwardS` is imho correct.

File: Documentation/index.rst (hunk 2 above)
Comment (Sebobo) [reply Documentation/index.rst]: Ah I see there is a merge error

Comment (Sebobo) [reply Documentation/index.rst]: Better now?

**Pull request merged**